Framework/API/src/Algorithm.cpp

```diff
@@ -1369,6 +1369,38 @@ bool Algorithm::doCallProcessGroups(
     const float duration = timer.elapsed();
     // Log that execution has completed.
     reportCompleted(duration, true /* this is for group processing*/);
+
+    m_history = boost::make_shared<AlgorithmHistory>(this, startTime, duration,
+                                                     ++g_execCount);
+
+    if (trackingHistory() && m_history) {
+
+      std::vector<Workspace_sptr> inputWorkspaces, outputWorkspaces;
+      findWorkspaceProperties(inputWorkspaces, outputWorkspaces);
+
+      // We need to find the workspaces to add the history to.
+      if (outputWorkspaces.size() == 0 && inputWorkspaces.size() == 0) {
+        for (auto ws : m_groupWorkspaces) {
```

Comment:
`const auto &`?

In fact, the `for` loop could be replaced by
```
outputWorkspaces.insert(outputWorkspaces.end(), m_groupWorkspaces.begin(), m_groupWorkspaces.end());
```

Reply:
:+1: